src/cython_cmake/cmake/FindCython.cmake

```diff
@@ -69,20 +76,39 @@ if(CYTHON_EXECUTABLE)
     set(_error_msg "Command \"${CYTHON_version_command}\" failed with")
     set(_error_msg "${_error_msg} output:\n${CYTHON_version_error}")
     message(SEND_ERROR "${_error_msg}")
+  elseif("${CYTHON_version_output}" MATCHES "^[Cc]ython version ([^,]+)")
+    set(CYTHON_VERSION "${CMAKE_MATCH_1}")
+  elseif("${CYTHON_version_error}" MATCHES "^[Cc]ython version ([^,]+)")
+    set(CYTHON_VERSION "${CMAKE_MATCH_1}")
   else()
-    if("${CYTHON_version_output}" MATCHES "^[Cc]ython version ([^,]+)")
-      set(CYTHON_VERSION "${CMAKE_MATCH_1}")
-    else()
-      if("${CYTHON_version_error}" MATCHES "^[Cc]ython version ([^,]+)")
-        set(CYTHON_VERSION "${CMAKE_MATCH_1}")
-      endif()
-    endif()
+    message(SEND_ERROR "Invalid Cython version output")
   endif()
 endif()
 
 include(FindPackageHandleStandardArgs)
-FIND_PACKAGE_HANDLE_STANDARD_ARGS(Cython REQUIRED_VARS CYTHON_EXECUTABLE)
 
-mark_as_advanced(CYTHON_EXECUTABLE)
+if(CMAKE_VERSION VERSION_LESS 3.19)
+  find_package_handle_standard_args(Cython
+    REQUIRED_VARS CYTHON_EXECUTABLE
+    VERSION_VAR ${CYTHON_VERSION}
+  )
+else()
+  find_package_handle_standard_args(Cython
+    REQUIRED_VARS CYTHON_EXECUTABLE
+    VERSION_VAR ${CYTHON_VERSION}
+    HANDLE_VERSION_RANGE
```

Comment:
Maybe just conditionally store `HANDLE_VERSION_RANGE` into a variable that can be expanded in the call to `find_package_handle_standard_args` so that you don't have to rewrite the rest of it.